website/src/utils/YamlData.js

```diff
@@ -102,10 +103,21 @@ export class YamlData {
     }
 
     delete(path) {
-        const data = JSON.parse(JSON.stringify(this.yaml))
+        let data = JSON.parse(JSON.stringify(this.yaml))
         const steps = path.split(".")
         const parent = getValueAtPath(data, steps.slice(0, -1));
-        parent.splice(steps.at(-1), 1);
+        if(path.includes('connections')){
```

Comment:
this code is specific to a connection. and it is inside the YamlData class. 
Haven't we said that specific handling is else where?